src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -8986,7 +8986,7 @@ private void VisitDeconstructMethodArguments(ArrayBuilder<DeconstructionVariable
                     // update the deconstruct method with any inferred type parameters of the containing type
                     if (deconstructMethod.OriginalDefinition != deconstructMethod)
                     {
-                        deconstructMethod = deconstructMethod.OriginalDefinition.AsMember((NamedTypeSymbol)rightResult.Type);
+                        deconstructMethod = (MethodSymbol)AsMemberOfType(rightResult.Type, deconstructMethod);
```

Comment:
:memo: This is what we do for nullability analysis of `BoundCall` (see `ReinferMethodAndVisitArguments`)